src/js/factory/weekView.js

```diff
@@ -31,52 +31,92 @@ var TimeResize = require('../handler/time/resize');
 var MilestoneClick = require('../handler/milestone/click');
 var TaskClick = require('../handler/task/click');
 
+var DEFAULT_VIEW_SEQUENCE = ['Milestone', 'Task', 'AllDay', 'TimeGrid'];
+var DEFAULT_VIEWS = {
+    'Milestone': {
+        minHeight: 20,
+        height: 80,
+        maxHeight: 80,
+        show: true
+    },
+    'Task': {
+        minHeight: 40,
+        height: 120,
+        maxHeight: 120,
+        show: true
+    },
+    'AllDay': {
+        minHeight: 20,
+        height: 80,
+        maxHeight: 80,
+        show: true
+    },
+    'TimeGrid': {
+        autoHeight: true,
+        show: true
+    }
+};
+
+/* eslint-disable complexity*/
 module.exports = function(baseController, layoutContainer, dragHandler, options) {
     var weekView, dayNameContainer, dayNameView, vLayoutContainer, vLayout,
         milestoneView, taskView, alldayView, timeGridView,
-        milestoneClickHandler, taskClickHandler, alldayClickHandler, alldayCreationHandler,
-        alldayMoveHandler, alldayResizeHandler, timeClickHandler, timeCreationHandler,
-        timeMoveHandler, timeResizeHandler, daynameClickHandler,
-        panels;
+        panels = [];
+    var viewSequence = options.week.viewSequence || DEFAULT_VIEW_SEQUENCE,
+        views = options.week.views || DEFAULT_VIEWS;
```

Comment:
Would you declare **panels, viewSequence, view** before _undefined_ variables?

Reply:
OK, I'll change it.